src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -13772,20 +13772,28 @@ void Unit::Kill(Unit* victim, bool durabilityLoss)
             Player* creditedPlayer = GetCharmerOrOwnerPlayerOrPlayerItself();
             /// @todo do instance binding anyway if the charmer/owner is offline
 
-            if (instanceMap->IsDungeon() && creditedPlayer)
+            if (instanceMap->IsDungeon() && (creditedPlayer || this == victim))
             {
                 if (instanceMap->IsRaidOrHeroicDungeon())
                 {
                     if (creature->GetCreatureTemplate()->flags_extra & CREATURE_FLAG_EXTRA_INSTANCE_BIND)
-                        ((InstanceMap*)instanceMap)->PermBindAllPlayers(creditedPlayer);
+                    {
+                        // if the boss killed itself we still need to bind players to the instance
+                        if (!creditedPlayer)
+                            creditedPlayer = ((InstanceMap*)instanceMap)->GetPlayers().getFirst()->GetSource();
```

Comment:
What if the instance becomes empty? Creatures can still update in grid if they are set to active